plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java

```diff
@@ -752,4 +752,158 @@ private void roundTrip(Class<?> c, Map<String,Object> m, String toString) throws
     private static void schema(Class<?> c, String schema) throws Exception {
         assertEquals(schema, new DescribableModel(c).toString());
     }
+
+    public static final class AllJavaStandardTypesClass {
+        // final values (set in constructor)
+        private final boolean booleanValue1;
+        private final byte byteValue1;
+        private final short shortValue1;
+        private final int intValue1;
+        private final long longValue1;
+        private final float floatValue1;
+        private final double doubleValue1;
+
+        // non-final values (set in setter)
+        @DataBoundSetter private boolean booleanValue2;
+        @DataBoundSetter private byte byteValue2;
+        @DataBoundSetter private short shortValue2;
+        @DataBoundSetter private int intValue2;
+        @DataBoundSetter private long longValue2;
+        @DataBoundSetter private float floatValue2;
+        @DataBoundSetter private double doubleValue2;
+
+        @DataBoundConstructor
+        public AllJavaStandardTypesClass(boolean booleanValue1, byte byteValue1, short shortValue1,
+                               int intValue1, long longValue1, float floatValue1, double doubleValue1) {
+            this.booleanValue1 = booleanValue1;
+            this.byteValue1 = byteValue1;
+            this.shortValue1 = shortValue1;
+            this.intValue1 = intValue1;
+            this.longValue1 = longValue1;
+            this.floatValue1 = floatValue1;
+            this.doubleValue1 = doubleValue1;
+        }
+
+        @DataBoundSetter public void setBooleanValue2(boolean booleanValue2) {
```

Comment:
This does not make sense. A given field may either be mentioned in a `@DataBoundConstructor`, or a `@DataBoundSetter`, but not both.